src/app/features/balances-list/balances-list.tsx

```diff
@@ -24,8 +28,7 @@ export const BalancesList = ({ address, ...props }: BalancesListProps) => {
     Object.keys(balances.fungible_tokens).length === 0 &&
     Object.keys(balances.non_fungible_tokens).length === 0;
 
-  if (noAssets && currentAccount?.address)
-    return <NoAssets address={currentAccount.address} {...props} />;
+  if (noAssets) return <NoAssets onFundAccount={handleFundAccount} />;
```

Comment:
As far as I can tell, this is where the test fails. Previously, we passed `{...props}` to no assets, which includes the `data-testid` selector. 

So, when we create a new account in test `should be able to create a new account then switch`, it loads with no balances, and the `waitForHomepage` event never fires, because the element doesn't appear. 

We should adjust the waitForHomepage logic

Reply:
Tests fixed. @fbwoolf maybe you can think of a more reliable way we can detect when we're on the home page, besides passing a prop down through two components?

Reply:
Interesting, ok, thanks for the fix.